cmd/client/main.go

```diff
@@ -211,6 +223,84 @@ func (c *Client) run(o *GrpcProxyClientOptions) error {
 }
 
 func (c *Client) getDialer(o *GrpcProxyClientOptions) (func(ctx context.Context, network, addr string) (net.Conn, error), error) {
+	if o.proxyUdsName != "" {
+		return c.getUDSDialer(o)
+	}
+	return c.getMTLSDialer(o)
+}
+
+func (c *Client) getUDSDialer(o *GrpcProxyClientOptions) (func(ctx context.Context, network, addr string) (net.Conn, error), error) {
```

Comment:
This has a lot of code duplication. To support UDS, for the grpc mode, we need to create a different dialOption, and for the http-Connect mode, we should use `net.Dial("unix", o.proxyUdsName)` instead of the `tls.Dial`. Can we just do those?

Reply:
While I agree there is a bunch of code duplication, there are also a lot of small differences, more that just the Dial method. I would suggest we merge what works and then look at trying to unify it later.